master/reference/calicoctl/setup/etcdv2.md

```diff
@@ -5,38 +5,80 @@ layout: docwithnav
 
 This document covers the configuration options for calicoctl when using an etcdv2 datastore.
 
+There are two ways to configure calicoctl with your etcdv2 cluster details: 
+configuration file or environment variables.
+
 ## Configuration file 
 
 By default `calicoctl` looks for a configuration file at `/etc/calico/calicoctl.cfg`.
 
 The file location may be overridden using the `--config` option on commands that required
 datastore access.
 
-The config file is a yaml map, with the following supported fields: 
+The config file is a yaml or json document in the following format:
+
+```
+apiVersion: v1
+kind: calicoApiConfig
+metadata:
+spec:
+  datastoreType: "etcdv2"
+  etcdEndpoints: "http://etcd1:2379,http://etcd2:2379"
+  ...
+```
+
+See table below for details on the etcdv2 specific fields that may be included in
+the spec section.
+
+If the file exists, then it must be valid and readable by calicoctl.  If the file
+does not exist, calicoctl will read access details from the environment variables.
 
-| Field           | Description                                | Examples
-|-----------------|--------------------------------------------|----------
-| etcdEndpoints   | A comma separated list of etcd endpoints.  | http://etcd1:2379 
-| etcdUsername    | Username for RBAC (optional)               | "user" 
-| etcdPassword    | Password for the given username (optional) | "password"
-| etcdKeyFile     | Path to the etcd key file (optional)       | /etc/calico/key.pem
-| etcdCertFile    | Path to the etcd client cert (optional)    | /etc/calico/cert.pem
-| etcdCACertFile  | Path to the etcd CA file (optional)        | /etc/calico/ca.pem
+## Environment variables
 
-> **NOTE** 
+If you are not using a config file to specify your access information, calicoctl
+will check a particular set of environment variables.
+
+See the table below for details on the etcdv2 specific environment variables.
+
+>  Note that if neither file nor environment variables are set, calicoctl defaults to
+>  using etcdv2 with a single endpoint of http://127.0.0.1:2379.
+
+## Complete list of etcdv2 connection configuration
+
+| Spec field      | Environment       | Description                                | Examples
+|-----------------|----------------------------------------------------------------|----------
+| datastoreType   | DATASTORE_TYPE    | Indicates the datastore to use (optional)  | etcdv2
```

Comment:
Should mention what the defaults are here (and probably for etcdEndpoints too: I'm guessing http://localhost:2379?)